pyht/async_client.py

```diff
@@ -348,32 +370,35 @@ async def _tts_grpc(
 
     async def _tts_http(
         self,
-        text: str | list[str],
+        text: Union[str, list[str]],
         options: TTSOptions,
-        voice_engine: str | None,
+        voice_engine: Optional[str],
         metrics: Metrics,
-        context: AsyncContext | None = None
+        streaming: bool = True,
+        context: Optional[AsyncContext] = None
     ) -> AsyncIterable[bytes]:
-        if voice_engine is None:
-            voice_engine = "Play3.0-mini-http"
-        if voice_engine == "Play3.0":
-            logging.warning("Voice engine Play3.0 is deprecated; use Play3.0-mini-http instead.")
-        elif voice_engine != "Play3.0-mini-http":
-            raise ValueError("Only Play3.0-mini-http is supported in the HTTP API; otherwise use the gRPC API.")
+        supported_voice_engines = ["Play3.0-mini", "PlayDialog"]
+        if voice_engine not in supported_voice_engines:
+            raise ValueError(f"Only {supported_voice_engines} are supported in the HTTP API; got {voice_engine}")
 
         start = time.perf_counter()
         await self.ensure_inference_coordinates()
+        assert self._inference_coordinates is not None, "No connection"
+
+        if streaming:
+            url = self._inference_coordinates[voice_engine]["http_streaming_url"]
+        else:
+            url = self._inference_coordinates[voice_engine]["http_nonstreaming_url"]
```

Comment:
Like mentioned before: best to use `.get(target, fallback)` here for guaranteeing data types.
```suggestion
            url = self._inference_coordinates.get(voice_engine, {}).get("http_streaming_url", "")
        else:
            url = self._inference_coordinates.get(voice_engine, {}).get("http_nonstreaming_url", "")
```

Reply:
See reply to previous comment; the bracket access is basically combining the get and the assert into one line, yeah?

Reply:
focused or ambiguous when erroring?